src/bootstrap/src/core/build_steps/dist.rs

```diff
@@ -2095,6 +2094,74 @@ pub fn maybe_install_llvm_runtime(builder: &Builder<'_>, target: TargetSelection
     }
 }
 
+/// Maybe add LLVM tools to the rustc sysroot.
+pub fn maybe_install_llvm_tools(builder: &Builder<'_>, target: TargetSelection, dst_dir: &Path) {
+    if builder.config.llvm_enabled(target) {
+        let llvm::LlvmResult { llvm_config, .. } = builder.ensure(llvm::Llvm { target });
+        if !builder.config.dry_run() && builder.config.llvm_tools_enabled {
+            let llvm_bin_dir =
+                command(llvm_config).arg("--bindir").run_capture_stdout(builder).stdout();
+            let llvm_bin_dir = Path::new(llvm_bin_dir.trim());
+
+            // Since we've already built the LLVM tools, install them to the sysroot.
+            // This is the equivalent of installing the `llvm-tools-preview` component via
+            // rustup, and lets developers use a locally built toolchain to
+            // build projects that expect llvm tools to be present in the sysroot
+            // (e.g. the `bootimage` crate).
+            for tool in LLVM_TOOLS {
+                let tool_exe = exe(tool, target);
+                let src_path = llvm_bin_dir.join(&tool_exe);
+                // When using `download-ci-llvm`, some of the tools
+                // may not exist, so skip trying to copy them.
+                if src_path.exists() {
+                    builder.copy_link(&src_path, &dst_dir.join(&tool_exe));
+                }
+            }
+        }
+    }
+}
+
+/// Maybe add `llvm-bitcode-linker` to the rustc sysroot.
+pub fn maybe_install_llvm_bitcode_linker(
+    builder: &Builder<'_>,
+    compiler: Compiler,
+    target: TargetSelection,
+    dst_dir: &Path,
+) {
+    if builder.config.llvm_bitcode_linker_enabled {
+        let dst_dir = dst_dir.join("self-contained");
+        t!(std::fs::create_dir_all(&dst_dir));
+
+        let src_path = builder.ensure(crate::core::build_steps::tool::LlvmBitcodeLinker {
+            compiler,
+            target,
+            extra_features: vec![],
+        });
+
+        let tool_exe = exe("llvm-bitcode-linker", target);
```

Comment:
Before, it was copied here:
```rust
 let tool_exe = exe("llvm-bitcode-linker", target_compiler.host);
builder.copy_link(&src_path, &libdir_bin.join(tool_exe));
```
Was it intended to change the location to be `<dst_dir>/self-contained` instead of `<dst_dir>`?

Reply:
It was intentional and adapted from `tool::LlvmBitcodeLinker` for consistency.

Reply:
I see. Ok :)